### DISCUSSION TITLE
Why is minizip creating zip files with paths using backslashes on Windows?

### DISCUSSION THREAD

**Discussion by strentler** [nmoinvaz/minizip#733] [Q&A]
Look at mz_os.h:
```
#define MZ_PATH_SLASH_UNIX              ('/')
#define MZ_PATH_SLASH_WINDOWS           ('\\')
#if defined(_WIN32)
#  define MZ_PATH_SLASH_PLATFORM        (MZ_PATH_SLASH_WINDOWS)
#else
#  define MZ_PATH_SLASH_PLATFORM        (MZ_PATH_SLASH_UNIX)
#endif
```

MZ_PATH_SLASH_PLATFORM is used to create metadata in the zip file which stores the paths with backslashes on windows instead of slashes.
I wonder why this is the case?
Also look at mz_zip.c:
```
       /* Ensure all slashes are written as forward slashes according to 4.4.17.1 */
        while ((err == MZ_OK) && (backslash = strrchr(next, '\\')) != NULL) {
            int32_t part_length = (int32_t)(backslash - next);

            if (mz_stream_write(stream, next, part_length) != part_length ||
                mz_stream_write(stream, "/", 1) != 1)
                err = MZ_WRITE_ERROR;

            left -= part_length + 1;
            next = backslash + 1;
        }
```

This changes the last backslash in a path to a slash. Independent of if running on windows or not.
This leads to metadata which has paths with mixed slashes and backslashes.
Also I wonder why this is done?

I also checked what happens when you use the system functionality of Windows 10 to create a zip file. Windows 10 is creating paths with slashes only. 

So I guess that the reason for backslashes for zip files created with minizip on windows must be something related to an old version of windows or some old tools?

The reason why I am asking this whole questions is because we use a C# unzipping library which is unable to unzip zip files which use backslashes for paths.

I wonder if minizip should be updated to only create zip files with paths having slashes only, or the C# unzipping library should be updated to also support backslashes?

**Comment by nmoinvaz** [comment]
It is likely a bug if that code does not conform to the comment above it. Feel free to submit a PR.

**Reply by nmoinvaz** [reply]
Btw, perhaps you have an old version, because most recent code uses `strchr` instead of `strrchr`. 
https://github.com/zlib-ng/minizip-ng/blob/680d6f1dcf9de99fc033b54975a1dfff10be2b6b/mz_zip.c#L795C10-L795C11
